galaxy_api/contrib/logging.py

```diff
@@ -0,0 +1,26 @@
+"""Logging utils."""
+
+import boto3
+import watchtower
+import logstash_formatter
+import request_id.logging
+from django.conf import settings
+
+
+class CloudWatchHandler(watchtower.CloudWatchLogHandler):
+    """Pre-configured CloudWatch handler."""
+
+    def __init__(self):
+        config = settings.CLOUD_WATCH_LOGGER
+        boto3_session = boto3.Session(
+            aws_access_key_id=config['AWS_ACCESS_KEY_ID'],
+            aws_secret_access_key=config['AWS_SECRET_ACCESS_KEY'],
+            region_name=config['AWS_REGION_NAME'],
+        )
+        super().__init__(
+            boto3_session=boto3_session,
+            log_group=config['LOGGING_GROUP'],
+            stream_name=config['LOGGING_STREAM_NAME'],
+        )
+        self.addFilter(request_id.logging.RequestIdFilter())
+        self.setFormatter(logstash_formatter.LogstashFormatterV1())
```

Comment:
Should be possible to composite the filter and formatter config via the configuration instead of in the class.
That would make the CloudWatchHandler setup more flexible via config, and reduce the dependencies needed by `galaxy_api.contrib.logging`.

Though it all depends on how the aws creds / boto session need to be setup if that is worthwhile.

ie

``` yaml
# log config yaml...
  formatters:
     # the existing formatters
     logstash:
        class: 'logstash_formatter.LogstashFormatterV1'
        # format could be '}' style as well if logstash_formatter supports it...
        format: '%(levelname)s %(asctime)s %(name) %(module)s:%(funcName)s:%(lineno)d %(process)d %(threadName)s %(message)s'
  filters:
     request_id: 
        '()': 'request_id.logging.RequestIdFilter'

  handlers:
    watchtower:
        formatter: logstash
        level: DEBUG
        (): galaxy_api.contrib.logging.CloudWatchHandler
        log_group: logger
        stream_name:  loggable
        filters:
          - request_id

```

Reply:
Usually more generic handler is preferable, but here the value is to provide specific pre-configured handler that can be simply enabled in logging config, comparing to variety of options in configuring and misconfiguring logging in config file.
Also please take into account that this is insights \ saas specific logger and it will be moved into a standalone project once we go on-prem.

For on-prem installation probably another more generic logger can be needed.